niceman/resource/session.py

```diff
@@ -194,49 +256,137 @@ def niceman_exec(self, command, args):
 
     @abc.abstractmethod
     def exists(self, path):
-        """Return if file exists"""
-        pass
+        """Determine if a path exists in the resource.
+        
+        Parameters
+        ----------
+        path : string
+            Path to check for existence in the resource environment.
+        
+        Returns
+        -------
+        bool
+            True if path exists
+        """
+        raise NotImplementedError
 
     @abc.abstractmethod
-    def put(self, src_path, dest_path, preserve_perms=False,
-                owner=None, group=None, recursive=False):
-        """Take file on the local file system and copy over into the session
+    def put(self, src_path, dest_path, uid=-1, gid=-1):
+        """Take file on the local file system and copy over into the resource
+        
+        Parameters
+        ----------
+        src_path : string
+            Path to file to push to resource environment
+        dest_path : string
+            Path of resource directory to put local file in
+        uid : number, optional
```

Comment:
where `number, ` came from?  https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt  does not mention it. Should be an `int` (can't be float, right?)